client/engine/engine.go

```diff
@@ -395,9 +399,8 @@ func (e *Engine) constructObjectiveFromMessage(id protocols.ObjectiveId, ss stat
 		if err != nil {
 			return &directdefund.Objective{}, fmt.Errorf("could not create direct defund objective from message: %w", err)
 		}
-
-		// TODO Destroy / Disable / Disown the associated ConsensusChannel
-
+		// If ddfo creation was successful, destroy the consensus channel to prevent it being used (a Channel will now take over governance)
+		e.store.DestroyConsensusChannel(ddfo.C.Id)
```

Comment:
... but we can't pull that trick here.

Reply:
We could make an issue to try and make the composite operation atomic in the store -- i.e. spawn one type and delete the other at the same time.

Reply:
I am in favour. If it is incorrect for a `channel.Channel` and a `consensus_channel.ConsensusChannel` with the same IDs to both exist, then SetConsensusChannel(c) itself should check / destroy a conflicting `channel.Channel`.

Lower-level, they could share the same namespace for keys, so that it's entirely impossible for both to exist.

Reply:
I'll merge this for now, since it gets the test assertions in place and an "ok" implementation. Filed #613